client/state/jetpack/credentials/reducer.js

```diff
@@ -63,6 +66,21 @@ export const updateRequestStatus = keyedReducer( 'siteId', ( state, { type } ) =
 	return state;
 } );
 
+export const testRequestStatus = keyedReducer( 'siteId', ( state, { type, role, testResult } ) => {
+	switch ( type ) {
+		case JETPACK_CREDENTIALS_TEST:
+			return { [ role ]: 'pending' };
+
+		case JETPACK_CREDENTIALS_TEST_SUCCESS:
+			return { [ role ]: testResult ? 'valid' : 'invalid' };
```

Comment:
Instead of storing a string value, should we just store a boolean `true` or `false`?

Reply:
Not possible as we need to handle 3 different states: `pending`, `valid` and `invalid`. I just omitted the `failure` state [based on this](https://github.com/Automattic/wp-calypso/pull/67589/files#r980698684).